cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java

```diff
@@ -288,4 +292,17 @@ private ProgramType getProgramType(String programType) throws BadRequestExceptio
       throw new BadRequestException(String.format("Invalid program type '%s'", programType), e);
     }
   }
+
+  /**
+   *
+   * @param namespaceId namespace ID
+   * @param appId application ID
+   * @return latest app version
+   */
+  private String getLatestAppVersion(String namespaceId, String appId) {
```

Comment:
nit: lets pass NamespaceId here instead of String?

Reply:
will change once the build is finished